### ISSUE TITLE
2 linear layer

### ISSUE THREAD

**Issue by nightfuryyy** [clovaai/deep-text-recognition-benchmark#225]
Thank you for awesome repo.
If i use Bi-LSTM and CTC, model will use 2 linear serial.
Is that correct ?
Thank you so much.

**Comment by lambdaterm** [comment]
Yes, if you use BiLSTM for sequence modeling, your model will contain two sequentially connected BiLSTM layers.

```python
        """ Sequence modeling"""
        if opt.SequenceModeling == 'BiLSTM':
            self.SequenceModeling = nn.Sequential(
                BidirectionalLSTM(self.FeatureExtraction_output, opt.hidden_size, opt.hidden_size),
                BidirectionalLSTM(opt.hidden_size, opt.hidden_size, opt.hidden_size))
            self.SequenceModeling_output = opt.hidden_size
        else:
            print('No SequenceModeling module specified')
            self.SequenceModeling_output = self.FeatureExtraction_output
```

https://github.com/clovaai/deep-text-recognition-benchmark/blob/master/model.py#L53

**Comment by nightfuryyy** [comment]
> Yes, if you use BiLSTM for sequence modeling, your model will contain two sequentially connected BiLSTM layers.
> 
> ```python
>         """ Sequence modeling"""
>         if opt.SequenceModeling == 'BiLSTM':
>             self.SequenceModeling = nn.Sequential(
>                 BidirectionalLSTM(self.FeatureExtraction_output, opt.hidden_size, opt.hidden_size),
>                 BidirectionalLSTM(opt.hidden_size, opt.hidden_size, opt.hidden_size))
>             self.SequenceModeling_output = opt.hidden_size
>         else:
>             print('No SequenceModeling module specified')
>             self.SequenceModeling_output = self.FeatureExtraction_output
> ```
> 
> https://github.com/clovaai/deep-text-recognition-benchmark/blob/master/model.py#L53
i mean this
https://github.com/clovaai/deep-text-recognition-benchmark/blob/master/modules/sequence_modeling.py#L9
https://github.com/clovaai/deep-text-recognition-benchmark/blob/master/model.py#L64

**Comment by lambdaterm** [comment]
You can use `print(model)` to see the model's structure. Hope, this helps.
```
DataParallel(
  (module): Model(
    (FeatureExtraction): ResNet_FeatureExtractor(
      // some details were skipped for compact view
    )
    (AdaptiveAvgPool): AdaptiveAvgPool2d(output_size=(None, 1))
    (SequenceModeling): Sequential(
      (0): BidirectionalLSTM(
        (rnn): LSTM(512, 256, batch_first=True, bidirectional=True)
        (linear): Linear(in_features=512, out_features=256, bias=True)
      )
      (1): BidirectionalLSTM(
        (rnn): LSTM(256, 256, batch_first=True, bidirectional=True)
        (linear): Linear(in_features=512, out_features=256, bias=True)
      )
    )
    (Prediction): Linear(in_features=256, out_features=37, bias=True)
  )
)
```

**Comment by nightfuryyy** [comment]
> You can use `print(model)` to see the model's structure. Hope, this helps.
> 
> ```
> DataParallel(
>   (module): Model(
>     (FeatureExtraction): ResNet_FeatureExtractor(
>       // some details were skipped for compact view
>     )
>     (AdaptiveAvgPool): AdaptiveAvgPool2d(output_size=(None, 1))
>     (SequenceModeling): Sequential(
>       (0): BidirectionalLSTM(
>         (rnn): LSTM(512, 256, batch_first=True, bidirectional=True)
>         (linear): Linear(in_features=512, out_features=256, bias=True)
>       )
>       (1): BidirectionalLSTM(
>         (rnn): LSTM(256, 256, batch_first=True, bidirectional=True)
>         (linear): Linear(in_features=512, out_features=256, bias=True)
>       )
>     )
>     (Prediction): Linear(in_features=256, out_features=37, bias=True)
>   )
> )
Thank you for this helps. 
But i saw that 2 linear serial but no activation or dropout between.
Is that ok ?
